### PR TITLE
Top recent trading pages

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #7603 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+export default compose(
+	whenActive( {
+		moduleName: 'analytics-4',
+		FallbackComponent: ConnectGA4CTATileWidget,
+	} ),
+	withCustomDimensions( {
+		dimensions:
+			KEY_METRICS_WIDGETS[ KM_ANALYTICS_TOP_RECENT_TRENDING_PAGES ]
+				.requiredCustomDimensions?.[ 0 ],
+		reportOptions: getReportOptions()[ 0 ],
```
Comment (tofumatt) [inline assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.js:206]: Oh, I see, this is why we have the function outside the component. That would be worth noting in a comment at the top of the file, as it's unclear otherwise and unexpected 😄 

Additionally, let's have the function return an object so the `reportOptions` and `dates` are keys rather than using numerical indexes for the returned values. `getReportOptions().reportOptions` is more readable than `getReportOptions()[ 0 ]`.

Even better might be `getReportOptions()` only returning the report options, given the dates are simple and consistent/hardcoded we could either specify them in both the function and the component, or have a separate `getDateRange()` function. 🤔

File: assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.stories.js [hunk 2]
```diff
@@ -26,12 +26,67 @@ import { KM_ANALYTICS_TOP_RECENT_TRENDING_PAGES } from '../../../../googlesiteki
...
+	orderby: [
+		{
+			metric: {
+				metricName: 'screenPageViews',
+			},
+			desc: true,
+		},
+	],
+	limit: 3,
+};
```
Comment (tofumatt) [inline assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.stories.js:74]: These should be exported from the component's file so the story can reuse them, rather than specifying them again here.

File: assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
+/**
+ * TopCitiesWidget component.
```
Comment (tofumatt) [inline assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.js:2]: This title doesn't match the filename/component name, by that raises a separate question: why rename this anyway? With this new filename it's the only KMW in that folder I can see whose component name/filename _doesn't_ end with `Widget`. I don't know what the motivation for that is, as I didn't see it in the ACs of the issue and it was mentioned in the IB but I'm not sure why.

cc @nfmohit who wrote the IB, maybe you can shed some light on the renaming here? 😄

**Review by tofumatt — CHANGES_REQUESTED** [review]
I think there's some duplication we can remove in the story file for this component, as well as some tweaks to the function that handles the report options.

I also have a question about the renaming, which I've pinged @nfmohit about as well, but if you know the reason @zutigrm just let me know 🙂

File: assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.js (hunk 3 above)
Comment (nfmohit) [reply assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.js:2]: Hi @tofumatt. The naming in the IB was an oversight. Since all widget names end with `Widget`, we should also do the same here. Thanks for checking!

**Comment by aaemnnosttv** [comment]
@zutigrm This branch also has merge conflicts that need resolving.

File: assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.stories.js (hunk 2 above)
Comment (zutigrm) [reply assets/js/modules/analytics-4/components/widgets/TopRecentTrendingPages.stories.js]: @tofumatt Great point, thanks. I extracted it into a new dir, so other custom dimensions dependant widgets can also use it.

File: assets/js/modules/index.test.js [hunk 4]
```diff
@@ -49,11 +49,17 @@ function getComponentNames( componentPath ) {
 describe( 'all modules', () => {
 	describe.each( directories( '.' ) )( '%s', ( moduleSlug ) => {
 		const components = directories( `${ moduleSlug }/components` );
-		if ( ! components.length ) {
+
+		// Filter out the custom-dimensions-report-options directory
+		const filteredComponents = components.filter(
```
Comment (techanvil) [inline assets/js/modules/index.test.js:54]: Hey @zutigrm, are these changes needed or could they have been introduced by accident? When I remove them and run the test suite locally, it still passes...

Comment (zutigrm) [reply assets/js/modules/index.test.js:54]: Hey @techanvil, they are not now, as the folder structure is removed in favour of date/report options functions being in the same file. I will remove that from the test in my next issue

Comment (techanvil) [reply assets/js/modules/index.test.js:54]: Sounds good, thanks @zutigrm!
